cc/src/core/light_epoch.h

```diff
@@ -176,44 +176,58 @@ class LightEpoch {
   /// Enter the thread into the protected code region
   inline uint64_t Protect() {
     uint32_t entry = Thread::id();
-    table_[entry].local_current_epoch = current_epoch.load();
-    return table_[entry].local_current_epoch;
+
+    auto e = current_epoch.load(std::memory_order_relaxed);
+    table_[entry].local_current_epoch.store(e, std::memory_order_relaxed);
+
+    std::atomic_thread_fence(std::memory_order_seq_cst);
+
+    return e;
   }
 
   /// Enter the thread into the protected code region
   /// Process entries in drain list if possible
   inline uint64_t ProtectAndDrain() {
     uint32_t entry = Thread::id();
-    table_[entry].local_current_epoch = current_epoch.load();
+    auto e = current_epoch.load(std::memory_order_relaxed);
```

Comment:
These three lines of code appears duplicated multiple times throughout the file. Consider putting them into a function?